backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/logging/impl/ElasticsearchSearchHints.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.elasticsearch.logging.impl;
+
+import org.hibernate.search.engine.backend.reporting.spi.BackendSearchHints;
+import org.hibernate.search.util.common.logging.impl.MessageConstants;
+
+import org.jboss.logging.Messages;
+import org.jboss.logging.annotations.Message;
+import org.jboss.logging.annotations.MessageBundle;
+
+@MessageBundle(projectCode = MessageConstants.PROJECT_CODE)
+public interface ElasticsearchSearchHints extends BackendSearchHints {
```

Comment:
Nice idea, TBH I wouldn't have bothered to explain *why* it's forbidden, I'd just have told users they can't do it.

BTW there's nothing backend-specific about most hints below... you could just move the backend-independent `@Message` annotations to `BackendSearchHints`?